tests/testing_tests/test_causal_test_suite.py

```diff
@@ -0,0 +1,132 @@
+        df["A"] = [1 if d > 50 else 0 for d in df["D"]]
+        df["C"] = df["D"] + (4 * (df["A"] + 2))  # C = (4*(A+2)) + D
+        self.observational_data_csv_path = os.path.join(temp_dir_path, "observational_data.csv")
+        df.to_csv(self.observational_data_csv_path, index=False)
+
+        self.causal_dag = CausalDAG(dag_dot_path)
+
+        # 3. Specify data structures required for test suite
+        self.expected_causal_effect = ExactValue(4)
+        test_list = [
+            CausalTestCase(
+                self.base_test_case,
+                self.expected_causal_effect,
+                0,
+                1,
+            ),
+            CausalTestCase(self.base_test_case, self.expected_causal_effect, 0, 2),
+        ]
+        self.estimators = [LinearRegressionEstimator]
+
+        # 3. Create test_suite and add a test
+        self.test_suite = CausalTestSuite()
+        self.test_suite.add_test_object(
+            base_test_case=self.base_test_case, causal_test_case_list=test_list, estimators=self.estimators
+        )
+
+    def test_adding_test_object(self):
+        "test an object can be added to the test_suite using the add_test_object function"
+        test_suite = CausalTestSuite()
+        test_list = [CausalTestCase(self.base_test_case, self.expected_causal_effect, 0, 1)]
+        estimators = [LinearRegressionEstimator]
+        test_suite.add_test_object(
+            base_test_case=self.base_test_case, causal_test_case_list=test_list, estimators=estimators
+        )
+        manual_test_object = {
+            self.base_test_case: {"tests": test_list, "estimators": estimators, "estimate_type": "ate"}
+        }
+        self.assertEqual(test_suite, manual_test_object)
+
+    def test_return_single_test_object(self):
+        """Test that a single test case can be returned from the test_suite"""
+        base_test_case = BaseTestCase(self.A, self.D)
+
+        test_list = [CausalTestCase(self.base_test_case, self.expected_causal_effect, 0, 1)]
+        estimators = [LinearRegressionEstimator]
+        self.test_suite.add_test_object(
+            base_test_case=base_test_case, causal_test_case_list=test_list, estimators=estimators
+        )
+
+        manual_test_case = {"tests": test_list, "estimators": estimators, "estimate_type": "ate"}
+
+        test_case = self.test_suite[base_test_case]
+
+        self.assertEqual(test_case, manual_test_case)
+
+    def test_execute_test_suite_single_base_test_case(self):
+        """Check that the test suite can return the correct results from dummy data for a single base_test-case"""
+        causal_test_engine = self.create_causal_test_engine()
+
+        causal_test_results = causal_test_engine.execute_test_suite(test_suite=self.test_suite)
+        causal_test_case_result = causal_test_results[self.base_test_case]
+        self.assertAlmostEqual(causal_test_case_result["LinearRegressionEstimator"][0].test_value.value, 4, delta=1e-10)
```

Comment:
How do we know the answer should be 4? i.e. is this a regression test - it was 4 last time we run it, or do we have an independent calculation?

Reply:
This test is a reimplementation of an existing `test_causal_test_engine` test case using the `causal_test_suite`:
https://github.com/CITCOM-project/CausalTestingFramework/blob/8d77deadc5e7ff819d0356fa1292e4cea07dbc59/tests/testing_tests/test_causal_test_engine.py#L159

So the source data and the result are identical to this test, so I'm not sure why 4 is the answer.

(Also feels a bit like a hitchhikers guide reference this)

Reply:
Fair enough - this make sense @jmafoster1, @AndrewC19 ?

Reply:
That makes sense. We synthesise the data on line 55 (`df["C"] = df["D"] + (4 * (df["A"] + 2))  # C = (4*(A+2)) + D`) such that the expected effect is 4.